e2e/cypress/integration/integrations/outgoing_webhook/disable_override_username_profile_spec.js

```diff
@@ -0,0 +1,643 @@
+
+        // # Save the settings
+        cy.get('#saveSetting').scrollIntoView().click({force: true});
+        cy.get('#saveSetting').should('be.disabled');
+    });
+
+    it('MM-T584 default username and profile pic Trigger = posting anything in the specified channel)', () => {
+        // # Enable user name and icon overrides
+        cy.apiAdminLogin();
+        enableUsernameAndIconOverride(true);
+
+        // # Define outgoing webhook
+        // * Go to test team/channel, open main menu and click "Integrations"
+        cy.visit(`${testTeam.name}/channels/${testChannel.name}`);
+        cy.get('#sidebarHeaderDropdownButton', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
+        cy.get('.Menu__content').should('be.visible').findByText('Integrations').click();
+
+        // * Verify that it redirects to integrations URL. Then, click "Outgoing Webhooks"
+        cy.url().should('include', `${testTeam.name}/integrations`);
+        cy.get('.backstage-sidebar').should('be.visible').findByText('Outgoing Webhooks').click();
+
+        // * Verify that it redirects to outgoing webhooks URL. Then, click "Add Outgoing Webhook"
+        cy.url().should('include', `${testTeam.name}/integrations/outgoing_webhooks`);
+        cy.findByText('Add Outgoing Webhook').click();
+
+        // * Verify that it redirects to where it can add outgoing webhook
+        cy.url().should('include', `${testTeam.name}/integrations/outgoing_webhooks/add`);
+
+        // # Enter webhook details such as title, description and channel, then save
+        cy.get('.backstage-form').should('be.visible').within(() => {
+            cy.get('#displayName').type('Webhook Title');
+            cy.get('#description').type('Webhook Description');
+            cy.get('#channelSelect').select(testChannel.display_name);
+            cy.get('#triggerWords').type(triggerWord);
+            cy.get('#callbackUrls').type(callbackUrl);
+            cy.findByText('Save').scrollIntoView().click();
+        });
+
+        // # Click "Done" and verify that it redirects to incoming webhooks URL
+        cy.findByText('Done').click();
+        cy.url().should('include', `${testTeam.name}/integrations/outgoing_webhooks`);
+
+        // # Click back to site name and verify that it redirects to test team/channel
+        cy.findByText(`Back to ${siteName}`).click();
+        cy.url().should('include', `${testTeam.name}/channels/${testChannel.name}`);
+
+        // # Post a message in the channel by user1
+        cy.apiLogin(testUser);
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.postMessage(triggerWordExpanded);
+        cy.uiWaitUntilMessagePostedIncludes(triggerWord);
+        cy.wait(TIMEOUTS.HALF_SEC);
+
+        cy.getLastPost().within(() => {
+            // * Verify that the user name is overriden
+            cy.get('.post__header').find('.user-popover').should('have.text', webhookUserNameOverride);
```

Comment:
I've got failing test here since the last post it's getting is not the one from the webhook. I suggests to wait for the message from webhook instead of explicit waiting. Please apply the same to other 8 similar instances.
```suggestion
        cy.uiWaitUntilMessagePostedIncludes('#### Outgoing Webhook Payload');

        cy.getLastPost().within(() => {
            // * Verify that the user name is overriden
            cy.get('.post__header').find('.user-popover').should('have.text', webhookUserNameOverride);
```